### DISCUSSION TITLE
`site.pages.find` doesn't behave as expected.

### DISCUSSION THREAD

**Discussion by ffflick-gh** [hexojs/hexo#4854] [Question / Answer]
Apparently, `site.pages` isn't really a native JS array, is it? I tried using `site.pages.find` but instead of a single page object it keeps returning a `_Query` object with a data field that contains ALL pages, instead of just returning the page I'm trying to find. I thought the variable was just a normal array of pages but it seems not, is this a bug or intended behaviour for whatever data type it is supposed to be?

**Comment by SukkaW** [comment]
> Apparently, `site.pages` isn't really a native JS array, is it?

No, it isn't.

>  I tried using `site.pages.find` but instead of a single page object it keeps returning a `_Query` object with a data field that contains ALL pages, instead of just returning the page I'm trying to find. I thought the variable was just a normal array of pages but it seems not, is this a bug or intended behaviour for whatever data type it is supposed to be?

http://hexojs.github.io/warehouse/
